src/main/java/DukeNUS.java

```diff
@@ -0,0 +1,142 @@
+            } catch (IndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The index of the task is missing or out of bounds.");
+            } catch (NumberFormatException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have entered is not a number.");
+            } catch (NullPointerException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have provided returns a null task.");
+            }
+            break;
+        case "unmark":
+            try {
+                Command.unmarkTask(tasks, parseInt(userInputWords[1]));
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The index of the task is missing or out of bounds.");
+            } catch (NumberFormatException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have entered is not a number.");
+            } catch (NullPointerException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have provided returns a null task.");
+            }
+            break;
+        case "todo":
+            try {
+                Command.addTodo(tasks, new Todo(userInputWords[1], false));
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: please specify the description of the todo.");
+            }
+            break;
+        case "deadline":
+            try {
+                String[] deadlineInput = userInputWords[1].split("/by", 2);
+                Command.addDeadline(tasks, new Deadline(deadlineInput[0], deadlineInput[1], false));
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: incorrect syntax. Correct usage: " +
+                        "`deadline [description] /by [deadline]`.");
+            } catch (DateTimeParseException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: invalid date format. Try the yyyy-MM-dd format (eg. 2000-12-31)");
+            }
+            break;
+        case "event":
+            try {
+                String[] eventInput = userInputWords[1].split("/from|/to", 3);
+                Command.addEvent(tasks, new Event(eventInput[0], eventInput[1], eventInput[2], false));
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: incorrect syntax. Correct usage: " +
+                        "`event [description] /from [start_date] /to [end_date]`.");
+            } catch (DateTimeParseException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: invalid date format. Try the yyyy-MM-dd format (eg. 2000-12-31)");
+            }
+            break;
+        default:
+            DukeNUSPrinter.printMessage("☹ Error: Unknown command. Please try again.");
+            break;
```

Comment:
This method is too long. Do consider seperating into sub-methods! Dont let a single method handle everything